[MSFT] Integration tests failure with `-DLLVM_ENABLE_REVERSE_ITERATION=On`

CI failed when running with DLLVM_ENABLE_REVERSE_ITERATION=On https://github.com/llvm/circt/actions/runs/24799105438/job/72576759892

Related to https://github.com/llvm/circt/issues/5660. 

```
"/usr/bin/python3.12" /home/runner/work/circt/circt/integration_test/Bindings/Python/dialects/msft.py 2> /home/runner/work/circt/circt/build/tools/circt/integration_test/Bindings/Python/dialects/Output/msft.py.tmp | /home/runner/work/circt/circt/build/bin/FileCheck /home/runner/work/circt/circt/integration_test/Bindings/Python/dialects/msft.py # RUN: at line 3
+ /usr/bin/python3.12 /home/runner/work/circt/circt/integration_test/Bindings/Python/dialects/msft.py
+ /home/runner/work/circt/circt/build/bin/FileCheck /home/runner/work/circt/circt/integration_test/Bindings/Python/dialects/msft.py
/home/runner/work/circt/circt/integration_test/Bindings/Python/dialects/msft.py:131:11: error: CHECK: expected string not found in input
 # CHECK: #msft.physloc<M20K, 2, 50, 1>
          ^
<stdin>:29:99: note: scanning from here
#msft.physloc<M20K, 2, 6, 1>, [#hw.innerNameRef<@top::@inst1>, #hw.innerNameRef<@MyWidget::@ext1>]
                                                                                                  ^
<stdin>:30:9: note: possible intended match here
=== Placements (col 2):
        ^
/home/runner/work/circt/circt/integration_test/Bindings/Python/dialects/msft.py:137:11: error: CHECK: expected string not found in input
 # CHECK: #msft.physloc<M20K, 2, 50, 1>
          ^
<stdin>:32:99: note: scanning from here
#msft.physloc<M20K, 2, 6, 1>, [#hw.innerNameRef<@top::@inst1>, #hw.innerNameRef<@MyWidget::@ext1>]
                                                                                                  ^
<stdin>:33:9: note: possible intended match here
=== Placements (col 2, row > 10):
        ^

Input file: <stdin>
Check file: /home/runner/work/circt/circt/integration_test/Bindings/Python/dialects/msft.py

-dump-input=help explains the following input dump.

Input was:
<<<<<<
             .
             .
             .
            24: #msft.physloc<M20K, 2, 6, 1> 
            25: #msft.physloc<M20K, 2, 50, 1> 
            26: === Placements: 
            27: #msft.physloc<M20K, 40, 40, 1>, (unoccupied) 
            28: #msft.physloc<M20K, 2, 50, 1>, (unoccupied) 
            29: #msft.physloc<M20K, 2, 6, 1>, [#hw.innerNameRef<@top::@inst1>, #hw.innerNameRef<@MyWidget::@ext1>] 
check:131'0                                                                                                       X error: no match found
            30: === Placements (col 2): 
check:131'0     ~~~~~~~~~~~~~~~~~~~~~~~
check:131'1             ?                possible intended match
            31: #msft.physloc<M20K, 2, 50, 1>, (unoccupied) 
            32: #msft.physloc<M20K, 2, 6, 1>, [#hw.innerNameRef<@top::@inst1>, #hw.innerNameRef<@MyWidget::@ext1>] 
check:137'0                                                                                                       X error: no match found
            33: === Placements (col 2, row > 10): 
check:137'0     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
check:137'1             ?                          possible intended match
            34: #msft.physloc<M20K, 2, 50, 1>, (unoccupied) 
            35: === Placements (col 6): 
            36: === Placements ASC, ASC: 
            37: #msft.physloc<M20K, 0, 0, 1>, (unoccupied) 
            38: #msft.physloc<M20K, 0, 0, 0>, (unoccupied) 
             .
             .
             .
>>>>>>

--
```

--- a/integration_test/Bindings/Python/dialects/msft.py
+++ b/integration_test/Bindings/Python/dialects/msft.py
@@ -127,14 +127,14 @@ with ir.Context() as ctx, ir.Location.unknown():
   seeded_pdb.walk_placements(print_placement)
   # CHECK-LABEL: === Placements:
   # TODO: #msft.physloc<M20K, 40, 40, 1>, [@tag]
-  # CHECK: #msft.physloc<M20K, 2, 6, 1>, [#hw.innerNameRef<@top::@inst1>, #hw.innerNameRef<@MyWidget::@ext1>]
-  # CHECK: #msft.physloc<M20K, 2, 50, 1>
+  # CHECK-DAG: #msft.physloc<M20K, 2, 6, 1>, [#hw.innerNameRef<@top::@inst1>, #hw.innerNameRef<@MyWidget::@ext1>]
+  # CHECK-DAG: #msft.physloc<M20K, 2, 50, 1>
 
   print("=== Placements (col 2):")
   seeded_pdb.walk_placements(print_placement, bounds=(2, 2, None, None))
   # CHECK-LABEL: === Placements (col 2):
-  # CHECK: #msft.physloc<M20K, 2, 6, 1>, [#hw.innerNameRef<@top::@inst1>, #hw.innerNameRef<@MyWidget::@ext1>]
-  # CHECK: #msft.physloc<M20K, 2, 50, 1>
+  # CHECK-DAG: #msft.physloc<M20K, 2, 6, 1>, [#hw.innerNameRef<@top::@inst1>, #hw.innerNameRef<@MyWidget::@ext1>]
+  # CHECK-DAG: #msft.physloc<M20K, 2, 50, 1>
 
   print("=== Placements (col 2, row > 10):")
   seeded_pdb.walk_placements(print_placement, bounds=(2, 2, 10, None))

